pyqtorch/gpsr.py

```diff
@@ -161,9 +158,61 @@ def single_gap_shift(
                 / (4 * torch.sin(spectral_gap * shift / 2))
             )
 
-        def multi_gap_shift(*args, **kwargs) -> Tensor:
-            """Implements multi gap PSR rule."""
-            raise NotImplementedError("Multi-gap is not yet supported.")
+        def multi_gap_shift(
+            param_name: str,
+            values: dict[str, Tensor],
+            spectral_gaps: Tensor,
+            shift_prefac: Tensor = torch.tensor(0.5),
+        ) -> Tensor:
+            """Implements multi gap PSR rule.
```

Comment:
```suggestion
            """Implement multi gap PSR rule.
```

Maybe also add the paper ref ?